src/main/java/com/sequenceiq/cloudbreak/service/stack/handler/UpdateInstancesRequestHandler.java

```diff
@@ -109,11 +110,15 @@ public Boolean call() throws Exception {
                                 List<Resource> resources = new ArrayList<>();
                                 for (final ResourceBuilder resourceBuilder : instanceResourceBuilders.get(cloudPlatform)) {
                                     CreateResourceRequest createResourceRequest =
-                                            resourceBuilder.buildCreateRequest(pCO, resources, resourceBuilder.buildResources(pCO, index, resources), index);
+                                            resourceBuilder.buildCreateRequest(pCO,
+                                                    resources,
+                                                    resourceBuilder.buildResources(pCO, index, resources, stack.getTemplateAsGroup(request.getHostGroup())),
+                                                    index,
+                                                    stack.getTemplateAsGroup(request.getHostGroup()));
                                     stackUpdater.addStackResources(stack.getId(), createResourceRequest.getBuildableResources());
                                     resources.addAll(createResourceRequest.getBuildableResources());
                                     resourceSet.addAll(createResourceRequest.getBuildableResources());
-                                    resourceBuilder.create(createResourceRequest);
+                                    resourceBuilder.create(createResourceRequest, stack.getTemplateSetAsList().get(0), stack.getRegion());
```

Comment:
Why is it needed to pass an instanceGroup to the create method? Isn't it part of the Resource that is part of the CreateResourceRequest? (And if it's needed for some reason why do we pass a random one here again?)